libcloud/compute/drivers/onapp.py

```diff
@@ -0,0 +1,540 @@
+                                format
+        :type       created_at: ``str``
+
+        :param      customer_network_id: the ID of the customer VLAN the IP
+                                         address belongs to
+        :type       customer_network_id: ``int``
+
+        :param      disallowed_primary: true if not allowed to be used as
+                                        primary (for VS build), otherwise false
+        :type       disallowed_primary: ``bool``
+
+        :param      free: true if free, otherwise false
+        :type       free: ``bool``
+
+        :param      gateway: gateway address
+        :type       gateway: ``str``
+
+        :param      hypervisor_id: the ID of a hypervisor the IP address is
+                                   associated with
+        :type       hypervisor_id: ``int``
+
+        :param      id: IP address id
+        :type       id: ``int``
+
+        :param      ip_address_pool_id: ID of the IP address pool the IP
+                                        address is associated with
+        :type       ip_address_pool_id: ``int``
+
+        :param      netmask: netmask for the IP address
+        :type       netmask: ``str``
+
+        :param      network_address: the address of the network
+        :type       network_address: ``str``
+
+        :param      network_id: the ID of the network
+        :type       network_id: ``int``
+
+        :param      pxe: true, if this hypervisor address can be used for
+                         cloudbooting a hypervisor
+        :type       pxe: ``bool``
+
+        :param      updated_at: the date when the network was updated in the
+                                    [YYYY][MM][DD]T[hh][mm][ss]Z format
+        :type       updated_at: ``str``
+
+        :param      user_id: the ID of the user this IP address is assigned to
+        :type       user_id: ``int``
+
+        :rtype:     :class:`OnAppIpAddress`
+        """
+        self.address = address
+        self.broadcast = broadcast
+        self.created_at = created_at
+        self.customer_network_id = customer_network_id
+        self.disallowed_primary = disallowed_primary
+        self.free = free
+        self.gateway = gateway
+        self.hypervisor_id = hypervisor_id
+        self.id = id
+        self.ip_address_pool_id = ip_address_pool_id
+        self.netmask = netmask
+        self.network_address = network_address
+        self.network_id = network_id
+        self.pxe = pxe
+        self.updated_at = updated_at
+        self.user_id = user_id
+
+    def __repr__(self):
+        return "<OnAppIpAddress>: id=%s, address=%s" % (self.id, self.address)
+
+
+class OnAppNodeDriver(NodeDriver):
+    """
+    Base OnApp node driver.
+    """
+
+    connectionCls = OnAppConnection
+    type = Provider.ONAPP
+    name = 'OnApp'
+    website = 'http://onapp.com/'
+
+    def create_node(self, label, memory, cpus, cpu_shares, hostname,
```

Comment:
This method is part of the standard API so all the arguments which are not part of standard API (looks like all of them :)) should have an `ex_` prefix.